lib/dialects/sqlite/query-generator.js

```diff
@@ -257,8 +257,11 @@ class SQLiteQueryGenerator extends MySqlQueryGenerator {
     return { query, bind };
   }
 
-  truncateTableQuery(tableName) {
-    return `DELETE FROM ${this.quoteTable(tableName)}`;
+  truncateTableQuery(tableName, options = {}) {
+    return [
+      `DELETE FROM ${this.quoteTable(tableName)}`,
+      options.restartIdentity ? `; DELETE FROM \`sqlite_sequence\` WHERE \`name\` = '${tableName.schema}.${tableName.table}';` : ''
```

Comment:
`this.quoteTable` for `sqlite_sequence`, `this.quoteIdentifier` for `name`

Reply:
`'${tableName.schema}.${tableName.table}'` should be replaced with `this.quoteTable(tableName)`

Reply:
I tried with quoteTable and it used backticks which wasn't valid, must be single inverted comma?

Reply:
Isn't delete from statement using same `quoteTable`, that works.

Reply:
From my testing it seemed that 

```
DELETE FROM `table`; DELETE FROM `sqlite_sequence` WHERE `name` = 'table'
```

Works but

```
DELETE FROM `table`; DELETE FROM `sqlite_sequence` WHERE `name` = `table`
```

Returns 

```
-- At line 1:
DELETE FROM `sqlite_sequence` WHERE `name` = `table`
-- Result: no such column: table
```

Reply:
Yeah that wont work as we are comparing against string `name`, you can try `'this.escape(tableName.table)'`. It is possible `tableName` is already a string, so you need to handle that as well

Do you think we need to prefix `schema` with table name? How is `sqlite_sequence` is storing table names, prefixed with schema?

Reply:
> 
> 
> Yeah that wont work as we are comparing against string `name`, you can try `'this.escape(tableName.table)'`. It is possible `tableName` is already a string, so you need to handle that as well
> 
> Do you think we need to prefix `schema` with table name? How is `sqlite_sequence` is storing table names, prefixed with schema?

Yes, `sqlite_sequence` name is `public.test_users` in the test examples.